src/pudl_archiver/archivers/eia/eia176.py

```diff
@@ -25,55 +39,86 @@ class Eia176Archiver(EiaNGQVArchiver):
-    items_url = "https://www.eia.gov/naturalgas/ngqs/data/items"
+    variables_url = "https://www.eia.gov/naturalgas/ngqs/data/items"
+    bulk_url = "https://www.eia.gov/naturalgas/ngqs/all_ng_data.zip"
 
-    async def get_items(self, url: str = items_url) -> list[str]:
-        """Get list of item codes from EIA NQGV portal."""
-        self.logger.info("Getting list of items in Form 176.")
-        items_response = await self.get_json(url)
-        items_list = [item["item"] for item in items_response]
-        return items_list
+    async def get_variables(self, url: str = variables_url) -> list[str]:
+        """Get list of variable codes from EIA NQGV portal.
+
+        Each variable code corresponds to a variable that we can download from the custom
+        report for EIA 176. We get all the variable codes
+        """
+        self.logger.info("Getting list of all variables in Form 176.")
+        variables_response = await self.get_json(url)
+        variables_list = [variable["item"] for variable in variables_response]
+        return variables_list
 
     async def get_resources(self) -> ArchiveAwaitable:
-        """Download EIA 176 resources."""
-        items_list = await self.get_items(url=self.items_url)
-        # Get the list of 176 datasets and their years of availability from the
-        # dataset list in the NGQV API. Unlike other NGQV datasets, we query the base
-        # URL only to get the list of years that data is available. The data itself is
-        # downloaded from the data_url.
-        datasets_list = await self.get_datasets(url=self.base_url, form=self.form)
-        dataset_years = set()
-        for dataset in datasets_list:
-            # Get all years for which there is 176 data available, based on information
-            # provided by the NGQV API.
-            dataset_years.update([year.ayear for year in dataset.available_years])
-        for year in dataset_years:
-            year = str(year)
-            yield self.get_year_resource(year, items_list=items_list)
-
-    async def get_year_resource(self, year: str, items_list: list[str]) -> ResourceInfo:
-        """Download all available data for a year.
-
-        We do this by getting the list of all items available for Form 176,
-        iteratively calling them into the URL, getting the JSON and transforming it into
-        a csv that is then zipped.
+        """Download EIA 176 resources, including all fields from the custom report."""
+        # First grab the bulk data
+        yield self.get_bulk_resource()
+
+        # Then archive the already-defined reports
+        reports_list = await self.get_reports(url=self.base_url, form=self.form)
+
+        # Check that we don't have any unexpected reports here
+        report_codes = {report.report_code for report in reports_list}
+        assert report_codes == {"RP1", "RP2", "RP3", "RP4", "RP5", "RP6", "RPC"}, (
+            f"Got unexpected reports for 176: {report_codes}."
```

Comment:
Small thing, but might be easier to parse this error if we only report the unexpected codes.

```suggestion
        report_codes = {report.report_code for report in reports_list}
        expected_report_codes = {"RP1", "RP2", "RP3", "RP4", "RP5", "RP6", "RPC"}
        assert report_codes == expected_report_codes, (
            f"Got unexpected reports for 176: {report_codes - expected_report_codes}."
```